### PR TITLE
Bach: add missing typings to Series methods

### PR DESCRIPTION
**Pull request by KathiaBarahona**

### CODE REVIEW

File: bach/bach/series/series.py [hunk 1]
```diff
@@ -1479,44 +1486,49 @@ def count(self, partition: WrappedPartition = None, skipna: bool = True):
...
         :param partition: The partition or window to apply
         :param skipna: only ``skipna=True`` supported. This means NULL values are ignored.
         :returns: a new Series with the aggregation applied
         """
-        return self._derived_agg_func(partition, 'max', skipna=skipna)
+        result = self._derived_agg_func(partition, 'max', skipna=skipna)
+        return cast(T, result)
 
-    def median(self, partition: WrappedPartition = None, skipna: bool = True):
+    def median(self: T, partition: WrappedPartition = None, skipna: bool = True) -> T:
```
Comment (thijs-obj) [inline bach/bach/series/series.py:1504]: Return type `T` matches what our code does, so this is correct in that sense.
This is actually different from what Pandas does tho. Added an issue to fix that later: https://github.com/objectiv/objectiv-analytics/issues/1042

File: bach/bach/series/series.py [hunk 2]
```diff
@@ -790,8 +789,8 @@ def to_numpy(self) -> numpy.ndarray:
 
     def materialize(
             self,
-            node_name='manual_materialize',
-            limit: Any = None,
+            node_name: str = 'manual_materialize',
+            limit: Optional[Any] = None,
```
Comment (thijs-obj) [inline bach/bach/series/series.py:793]:
```suggestion
            limit: Optional[Union[int, slice]] = None,
```

Comment (KathiaBarahona) [reply bach/bach/series/series.py]: Done here https://github.com/objectiv/objectiv-analytics/pull/1038/commits/92c23d3a6faeb51925458b5a169fe19bfbedb36f

File: bach/bach/series/series.py [hunk 3]
```diff
@@ -1089,7 +1088,7 @@ def notnull(self) -> 'SeriesBoolean':
         from bach import SeriesBoolean
         return self.copy_override_type(SeriesBoolean).copy_override(expression=expression)
 
-    def fillna(self, other: AllSupportedLiteralTypes):
+    def fillna(self, other: AllSupportedLiteralTypes) -> 'Series':
```
[the remark this answers is not in the document]
Comment (KathiaBarahona) [reply bach/bach/series/series.py]: Done here https://github.com/objectiv/objectiv-analytics/pull/1038/commits/92c23d3a6faeb51925458b5a169fe19bfbedb36f

**Pull request merged**